src/lib.rs

```diff
@@ -110,63 +124,107 @@ impl<L: LogLevel> Verbosity<L> {
     /// Get the log level.
     ///
     /// `None` means all output is disabled.
-    pub fn log_level(&self) -> Option<Level> {
-        level_enum(self.verbosity())
+    #[cfg(feature = "log")]
+    pub fn log_level(&self) -> Option<log::Level> {
+        self.filter().into()
     }
 
     /// Get the log level filter.
-    pub fn log_level_filter(&self) -> LevelFilter {
-        level_enum(self.verbosity())
-            .map(|l| l.to_level_filter())
-            .unwrap_or(LevelFilter::Off)
+    #[cfg(feature = "log")]
+    pub fn log_level_filter(&self) -> log::LevelFilter {
+        self.filter().into()
+    }
+
+    /// Get the trace level.
+    ///
+    /// `None` means all output is disabled.
+    #[cfg(feature = "tracing")]
+    pub fn tracing_level(&self) -> Option<tracing_core::Level> {
+        self.filter().into()
+    }
+
+    /// Get the trace level filter.
+    #[cfg(feature = "tracing")]
+    pub fn tracing_level_filter(&self) -> tracing_core::LevelFilter {
+        self.filter().into()
     }
 
     /// If the user requested complete silence (i.e. not just no-logging).
     pub fn is_silent(&self) -> bool {
-        self.log_level().is_none()
+        self.filter() == Filter::Off
     }
 
-    fn verbosity(&self) -> u8 {
-        let default_verbosity = level_value(L::default());
-        let verbosity = default_verbosity as i16 - self.quiet as i16 + self.verbose as i16;
-        verbosity.clamp(0, u8::MAX as i16) as u8
+    /// Get the filter level after applying `--verbose` and `--quiet`.
+    pub fn filter(&self) -> Filter {
+        L::default().with_offset(self.verbose as i16 - self.quiet as i16)
     }
 }
 
-fn level_value(level: Option<Level>) -> u8 {
-    match level {
-        None => 0,
-        Some(Level::Error) => 1,
-        Some(Level::Warn) => 2,
-        Some(Level::Info) => 3,
-        Some(Level::Debug) => 4,
-        Some(Level::Trace) => 5,
-    }
+/// A representation of the log level filter.
+///
+/// Used to calculate the log level and filter.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Filter {
```

Comment:
Do you think this should be named `LevelFilter`?

I avoided doing so as this crate previously re-exported `log::LevelFilter` at the root, so this might be a bit of a confusing change.

Reply:
Good point.  This crate is focused around "verbosity" so your previous idea of having `Verbosity` in the name works.  I suspect we should be consistent with the other crates in calling this a `Filter` but a part of me finds that weird and would prefer `Level` especially since we only have "what level to show" and not "what level is this record" and have to deal with distinguishing them.

Reply:
The filter part of the name communicates that `Off` is valid. Where-as there is no `Off` level that logs / trace events can be written at. I think Filter definitely belongs in the name because of this, it's mostly whether a prefix makes sense. I moved away from VerbosityFilter as it seemed like unnecessary repetition, but I think you might be right that it helps distinguish the filter as being distinctly this crate's Filter type.

Reply:
in #125 this became VerbosityFilter